app/javascript/elm/src/Data/Overlay.elm

```diff
@@ -0,0 +1,85 @@
+module Data.Overlay exposing (Model, Operation(..), Overlay(..), init, none, update, view)
+
+import Html exposing (Html, div, p, text)
+import Html.Attributes exposing (class, id)
+import List.Nonempty as Nonempty exposing (Nonempty(..))
+
+
+type Model
+    = Overlays (Nonempty Overlay)
+    | NoOverlay
```

Comment:
So this is equivalent to `type Model = Overlays (List Overlay)` where `NoOverlay` covers the case of an empty list, right? Can you share what are the advantages of that?
Cause I can see that having a normal list would make the updated shorter - you would only need @ branches - for adding and filtering the list. 
And in the view, you can just have `case (head model) of`.

Reply:
Good point. Removed nonempty but kept the opaque type. Thanks!